data_processing/meteo/pg_processing/task_functions.py

```diff
@@ -0,0 +1,286 @@
+        dates = [item for item in dates if item > latest_processed_date]
+        for dataset in DATASETS_TO_PROCESS:
+            resources = fetch_resources(dataset)
+            mydict.update(process_resources(resources, dataset, latest_ftp_processing, dates))
+        ti.xcom_push(key="latest_processed_date", value=max(dates))
+
+    ti.xcom_push(key="regex_infos", value=mydict)
+
+
+
+def unzip_csv_gz():
+    for dataset in DATASETS_TO_PROCESS:
+        patterns = []
+        if dataset == "BASE/QUOT":
+            pattern_vent = os.path.join(DATADIR, config[dataset]["table_name"] + "_vent", '*.csv.gz')
+            pattern_autres = os.path.join(DATADIR, config[dataset]["table_name"] + "_autres", '*.csv.gz')
+            patterns.extend([pattern_vent, pattern_autres])
+        else:
+            pattern = os.path.join(DATADIR, config[dataset]["table_name"], '*.csv.gz')
+            patterns.append(pattern)
+
+        for pattern in patterns:
+            for file_path in glob.glob(pattern):
+                output_file_path = file_path[:-3]
+
+                with gzip.open(file_path, 'rb') as f_in:
+                    with open(output_file_path, 'wb') as f_out:
+                        shutil.copyfileobj(f_in, f_out)
+                print(f'Unzipped {file_path} to {output_file_path}')
+
+
+def delete_and_insert_into_pg(ti):
+    regex_infos = ti.xcom_pull(key="regex_infos", task_ids="download_data")
+    for dataset in DATASETS_TO_PROCESS:
+        tables = []
+        if dataset == "BASE/QUOT":
+            tables.append("base_quot_vent")
+            tables.append("base_quot_autres")
+        else:
+            tables.append(config[dataset]["table_name"])
+        for table in tables:
+            print(table)
+            csv_paths = glob.glob(DATADIR + table + "/*.csv")
+
+            db_params = {
+                'dbname': conn.schema,
+                'user': conn.login,
+                'password': conn.password,
+                'host': conn.host,
+                'port': conn.port,
+            }
+
+            try:
+                conn2 = psycopg2.connect(**db_params)
+                conn2.autocommit = False  # Disable autocommit to start a transaction
+                delete_old_data(conn2, table, regex_infos)
+                for csv_path in csv_paths:
+                    print(csv_path)
+                    load_csv_to_postgres(conn2, csv_path, table, regex_infos)
+
+                # Commit the transaction
+                conn2.commit()
+                print("Transaction committed successfully.")
+
+            except Exception as e:
+                # Rollback the transaction in case of error
+                conn2.rollback()
+                print(f"An error occurred: {e}")
+                print("Transaction rolled back.")
+                raise
+
+            finally:
+                conn2.close()
+
+def load_csv_to_postgres(conn, csv_path, table_name, regex_infos):
```

Comment:
Is that too custom to use [our beautiful util](https://github.com/datagouv/datagouvfr_data_pipelines/blob/4fba2fa0efc6bb5d43cdc5d2774fcbb07dc87f86/utils/postgres.py#L149)? (or maybe we can upgrade it)

Reply:
no because i need to do all that in a transaction and for now it is not possible with this beautiful util